python/src/trezorlib/_internal/firmware_headers.py

```
@@ -487,17 +495,26 @@ def verify(self, dev_keys: bool = False) -> None:
+        if mask.bit_length() > len(self.public_ec_keys(dev_keys)):
+            raise ValueError("Sigmask specifies more public keys than provided.")
+
+        for i, key in enumerate(self.public_ec_keys(dev_keys)):
+            if not (mask & (1 << i)):
+                continue
+            ext_digest = hash_fn(digest + self.unauth.slh_signatures[i]).digest()
             try:
-                _ed25519.checkvalid(self.unauth.ec_signatures[idx], ext_digest, key)
+                _ed25519.checkvalid(self.unauth.ec_signatures[i], ext_digest, key)
             except _ed25519.SignatureMismatch:
                 raise firmware.InvalidSignatureError("Invalid bootloader signature")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)